### PR TITLE
growing index memory opt & getvector bugfix

### PR DESCRIPTION
**Pull request by foxspy**
#25228

### CODE REVIEW

File: internal/core/src/segcore/FieldIndexing.cpp [hunk 1]
```diff
@@ -122,13 +125,11 @@ VectorFieldIndexing::AppendSegmentIndex(int64_t reserved_offset,
         }
         auto dataset = knowhere::GenDataSet(vec_num, dim, data_addr);
         dataset->SetIsOwner(false);
-        auto indexing = std::make_unique<index::VectorMemIndex>(
-            config_->GetIndexType(), config_->GetMetricType());
-        indexing->BuildWithDataset(dataset, conf);
+        index_->BuildWithDataset(dataset, conf);
```
Comment (czs007) [inline internal/core/src/segcore/FieldIndexing.cpp:128]: should exception-safe

File: internal/core/src/segcore/FieldIndexing.h [hunk 2]
```diff
@@ -323,6 +335,16 @@ class IndexingRecord {
         }
         return false;
     }
+
+    bool
+    IndexHoldRawData(FieldId fieldId) const {
+        if (is_in(fieldId)) {
```
Comment (czs007) [inline internal/core/src/segcore/FieldIndexing.h:341]: rename to HasRawData

File: internal/core/src/segcore/SegmentGrowingImpl.cpp [hunk 3]
```diff
@@ -406,6 +426,19 @@ SegmentGrowingImpl::bulk_subscript_impl(FieldId field_id,
...
+    if (HasRawData(field_id.get())) {
+        //When data sync with index
+        if (indexing_record_.SyncDataWithIndex(field_id)) {
+            indexing_record_.GetDataFromIndex(
+                field_id, seg_offsets, count, element_sizeof, output_raw);
+        } else {
+            //Else copy from chunk
+            std::lock_guard<std::shared_mutex> guard(chunk_mutex_);
+            copy_from_chunk();
+        }
```
Comment (czs007) [inline internal/core/src/segcore/SegmentGrowingImpl.cpp:441]: add AssertInfo here.

Comment (foxspy) [reply internal/core/src/segcore/SegmentGrowingImpl.cpp:435]: done

File: internal/core/src/segcore/FieldIndexing.h (hunk 2 above)
Comment (foxspy) [reply internal/core/src/segcore/FieldIndexing.h:338]: done

File: internal/core/src/segcore/FieldIndexing.cpp (hunk 1 above)
Comment (foxspy) [reply internal/core/src/segcore/FieldIndexing.cpp]: Good! Done

**Pull request merged**